File: s03_Java/RunnableHandler/app/src/main/java/com/omeryildizce/runnablehandler/MainActivity.java
package com.omeryildizce.runnablehandler;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView textView;
    int second;
    int minute;
    int hour;
    Runnable runnable;
    Handler handler;
    Button buttonStart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        second = 0;
        minute = 0;
        hour = 0;
        buttonStart = findViewById(R.id.buttonStart);
    }

    public void start(View view) {
        handler = new Handler();
        runnable = new Runnable() {
            @Override
            public void run() {
                textView.setText(String.format("Time: %02d:%02d:%02d", hour, minute, second));
                second++;
                if (second % 60 == 0) {
                    minute++;
                    second = 0;
                    if (minute % 60 == 0) {
                        hour++;
                        minute = 0;
                    }
                }
                handler.postDelayed(runnable, 1_000);
            }
        };
        handler.post(runnable);
        buttonStart.setEnabled(false);
    }

    public void stop(View view) {
        buttonStart.setEnabled(true);
        handler.removeCallbacks(runnable);
        second = 0;
        minute = 0;
        hour = 0;
    }
}

File: s16_Kotlin/Runnables/app/src/main/java/com/omeryildizce/runnables/MainActivity.kt
package com.omeryildizce.runnables

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import com.omeryildizce.runnables.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var number = 0
    var runnable: Runnable = Runnable { }
    var handler: Handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.start.setOnClickListener {
            number = 0
            runnable = object : Runnable {
                override fun run() {
                    number++
                    binding.time.setText(String.format("%02d:%02d:%02d", (number /3600) % 99, (number /60) % 60 , number % 60))

                    handler.postDelayed(this, 1000)
                }
            }
            handler.post(runnable)
        }
        binding.stop.setOnClickListener {
            handler.removeCallbacks(runnable)
            Toast.makeText(this,binding.time.text.toString(), Toast.LENGTH_LONG).show()
            number = 0
            binding.time.setText("00:00:00")
        }

    }
}